### PR TITLE
Fixed internal error related to ecrecover and ABIEncoderV2.

### LINKED ISSUE #5997: InternalCompilerError when upgrading to solidity 0.5.4
The following code works just fine with `solc 0.5.3`

```
pragma solidity ^0.5.3;
pragma experimental ABIEncoderV2;

library ECDSALib
{
	function recover(bytes32 hash, bytes memory sign)
	public pure returns (address)
	{
		bytes32 r;
		bytes32 s;
		uint8   v;
		require(sign.length == 65);
		assembly
		{
			r :=         mload(add(sign, 0x20))
			s :=         mload(add(sign, 0x40))
			v := byte(0, mload(add(sign, 0x60)))
		}
		if (v < 27) v += 27;
		require(v == 27 || v == 28);
		return ecrecover(hash, v, r, s);
	}
}
```

However, when moving to `solc 0.5.4` (other parts of my code could benefit from calldata structs) I get an error:

`InternalCompilerError: Non-padded and in-place encoding can only be combined.`

Apparently its the `ecrecover` call that fails, if I comment it and replace it with `return address(0)` the compilation passes.

### PR DESCRIPTION
**Pull request by chriseth**
Fixes #5997

### CODE REVIEW

File: libsolidity/codegen/ExpressionCompiler.cpp [hunk 1]
```diff
@@ -1965,11 +1965,16 @@ void ExpressionCompiler::appendExternalFunctionCall(
 	// If the function takes arbitrary parameters or is a bare call, copy dynamic length data in place.
 	// Move arguments to memory, will not update the free memory pointer (but will update the memory
 	// pointer on the stack).
+	bool encodeInPlace = _functionType.takesArbitraryParameters() || _functionType.isBareCall();
+	if (_functionType.kind() == FunctionType::Kind::ECRecover)
+		// This would the only combination of padding and in-place encoding,
```
Comment (leonardoalt) [inline libsolidity/codegen/ExpressionCompiler.cpp]:
```suggestion
		// This would be the only combination of padding and in-place encoding,
```

**Review by leonardoalt — DISMISSED** [review]
lgtm if tests work

**Review by chriseth — COMMENTED** [review]
I think this still needs more fixes, at least https://github.com/ethereum/solidity/pull/5102 has some failures that look similar.

**Comment by chriseth** [comment]
It turns out there was only a single other problem and it is a little bit different. Will create a new PR for it.

**Review by leonardoalt — DISMISSED** [review]

**Pull request merged**